### PR TITLE
fix: handing corrupt _txn file in Mig620

### PR DESCRIPTION
**Pull request by ideoma**

### CODE REVIEW

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 8 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/mig/Mig620.java|6|8|75.00%|

File: core/src/main/java/io/questdb/cairo/mig/Mig620.java [hunk 1]
```diff
@@ -188,11 +185,20 @@ private static LongList readColumnTopsAllPartitions(int columnCount, int partiti
...
         long offset = partitionSizeOffset + 4;
+        long prevPartition = Long.MIN_VALUE;
+        long txSize = txMemory.size() - 4 * 8;
         for (int partitionIndex = 0; partitionIndex < partitionCount; partitionIndex++) {
+            if (offset > txSize) {
+                throw CairoException.instance(0).put("corrupt _txn file ").put(path.trimTo(pathLen)).put(", file is too small to read offset ").put(offset);
+            }
             long partitionTs = txMemory.getLong(offset);
+            if (partitionTs <= prevPartition) {
+                throw CairoException.instance(0).put("corrupt _txn file, partitions are not ordered at ").put(path.trimTo(pathLen));
```
Comment (bluestreak01) [inline core/src/main/java/io/questdb/cairo/mig/Mig620.java:196]: path.trimTo() does not ensure string is zero-terminated

**Review by bluestreak01 — CHANGES_REQUESTED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 8 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/mig/Mig620.java|6|8|75.00%|

**Review by bluestreak01 — APPROVED** [review]

**Comment by ideoma** [comment]
#### [PR Coverage check]

:heart_eyes: **pass** : 6 / 8 (75.00%)




#### file detail

|   |path|covered line|new line|coverage|
|----|----|----|----|----|
|:large_blue_circle:|io/questdb/cairo/mig/Mig620.java|6|8|75.00%|

**Pull request merged**